src/Akka.Streams.Kafka/Stages/Consumers/Abstract/BaseSingleSourceLogic.cs

```diff
@@ -118,45 +118,35 @@ protected virtual void MessageHandling((IActorRef, object) args)
                         _buffer.Enqueue(consumerMessage);
                     
                     Pump();
-                    
                     break;
                 
                 case Status.Failure failure:
-                    var exception = failure.Cause;
-                    switch (_decider(failure.Cause))
-                    {
-                        case Directive.Stop:
-                            // Throw
-                            FailStage(exception);
-                            break;
-                        case Directive.Resume:
-                            // keep going
-                            break;
-                        case Directive.Restart:
-                            // TODO: Need to do something here: https://github.com/akkadotnet/Akka.Streams.Kafka/issues/33
-                            break;
-                    }
+                    FailStage(failure.Cause);
                     break;
                 
-                case Terminated terminated:
+                case Terminated terminated when ReferenceEquals(terminated.ActorRef, ConsumerActor):
                     FailStage(new ConsumerFailed());
                     break;
             }
         }
 
         private void Pump()
         {
-            if (IsAvailable(_shape.Outlet))
+            while (true)
```

Comment:
This will conflict with https://github.com/akkadotnet/Akka.Streams.Kafka/pull/241

Reply:
i'll revert it then